ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -184,6 +187,17 @@ type FcsErrorsStageProcessBase(fsFile, daemonProcess) =
                     | _ -> TypeEquationError(expectedType, actualType, expr, error.Message) :> _
                 else null
 
+            | Regex typeDoesNotMatchMessage [expectedType; actualType] ->
+                let expr = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null)
+                TypeDoesNotMatchTypeError(expectedType, actualType, expr, error.Message)
+
+            | Regex ifBranchSatisfyContextTypeRequirements [expectedType; actualType] ->
+                let expr = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null)
+                IfExpressionNeedsTypeToSatisfyTypeRequirementsError(expectedType, actualType, expr, error.Message)
+
+            | Regex typeMisMatchTupleLengths [expectedType; actualType] ->
+                let expr = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null)
+                TypeMisMatchTuplesHaveDifferingLengthsError(expectedType, actualType, expr, error.Message)
```

Comment:
There's something wrong with reported types when item count mismatch:

```fsharp
let x: int * int * int = 1, ""
```

<img width="473" alt="Screenshot 2022-05-24 at 17 23 33" src="https://user-images.githubusercontent.com/3923587/170073582-923fbbc2-d5af-4d4e-838b-df3529190dab.png">

Let's disable this case until the error is fixed?